hedera-node/docs/design/services/smart-contract-service/scheduled-create-token-system-contract.md

```diff
@@ -0,0 +1,130 @@
+# Scheduled Token Create System Contract
+
+## Purpose
+
+[HIP-755](https://hips.hedera.com/hip/hip-755) defines a new system contract that allows accounts and contracts to interact with the schedule service system contract enabling the signing of scheduled transactions.
+[HIP-756](https://hips.hedera.com/hip/hip-756) to define new system contract functions to schedule token creation in order for a smart contract to be able to designate accounts for autoRenew and/or treasury roles.
+This document will define the architecture and implementation of this functionality.
+
+## Goals
+
+- Provide the users with the ability to schedule token creation and updates.
+- Provide the ability to retrieve the information about a scheduled creation/update.
+- ~~Provide the ability to retrieve the newly created token address via scheduled transaction.~~
+
+## Non Goals
+
+- The implementation of the Schedule Service logic as it is already done
+- The implementation of the HSSSystemContract infrastructure wiring as it is already done with HIP-755
+- The modification of existing Scheduled Transactions like updating or deleting is out of scope for this HIP.
+- The implementation of a more generic `scheduleContractCall` to enable cron like functionality for calling contract functions.
+While this functionality will not be explicitly designed in this document, care will be taken to ensure that the design is extensible to support this in the future.
+- Provide the ability to retrieve the newly created token address via scheduled transaction.  
+
+## Architecture and Implementation
+
+The architecture for the scheduled creations follows the existing framework defined for handling all calls to the HederaScheduleService system contract in the modularization services and is described in more detail in the section below.
+One difference is that instead of creating multiple functions for each type of token creation, we will have a single function called `scheduleNative` that will be able to handle all types of token creations.  This function will
```

Comment:
```suggestion
One difference is that instead of creating multiple functions for each type of HTS system contract functions, we will have a single function called `scheduleNative` that will be able to handle all types of HTS functions, starting with token creation and updates.  This function will
```

Reply:
Done